couler/core/config.py

```diff
@@ -38,7 +53,7 @@ def config_workflow(
     :return:
     """
     if name is not None:
-        states.workflow.name = name
+        states.workflow.name = states.workflow_name_salter(name)
```

Comment:
What if the name salter is None?

Reply:
There's a default implementation. and generally speaking we want to avoid name clashes so it's a design decision whether we want to `None` to work or not. I would say no, since differentiating between different runs of workflows by name is important IMO, but making `None` work like the identity function has its merits nonetheless.